packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -601,36 +601,80 @@ class _CupertinoContextMenuState extends State<CupertinoContextMenu> with Ticker
   }
 }
 
-// A floating copy of the CupertinoContextMenu's child.
-//
-// When the child is pressed, but before the CupertinoContextMenu opens, it does
-// an animation where it slowly grows. This is implemented by hiding the
-// original child and placing _DecoyChild on top of it in an Overlay. The use of
-// an Overlay allows the _DecoyChild to appear on top of siblings of the
-// original child.
-class _DecoyChild extends StatefulWidget {
-  const _DecoyChild({
+
+/// A floating copy of the CupertinoContextMenu's child.
+///
+/// When the child is pressed, but before the CupertinoContextMenu opens, it does
+/// an animation where it slowly grows. This is implemented by hiding the
+/// original child and placing DecoyChild on top of it in an Overlay. The use of
+/// an Overlay allows the DecoyChild to appear on top of siblings of the
+/// original child.
+class DecoyChild extends StatefulWidget {
+  /// Create a DecoyChild.
+  const DecoyChild({
+    super.key,
     this.beginRect,
     required this.controller,
     this.endRect,
     this.child,
     this.builder,
   });
 
+  /// The rect of the child at the moment that the CupertinoContextMenu opens.
   final Rect? beginRect;
+
+  /// The controller that drives the animation of the decoy child.
   final AnimationController controller;
+
+  /// The ending rectangle of the context menu decoy child.
   final Rect? endRect;
+
+  /// The child widget that is being previewed.
   final Widget? child;
+
+  /// A function that builds the child and handles the transition between the
+  /// default child and the preview when the CupertinoContextMenu is open.
   final CupertinoContextMenuBuilder? builder;
 
   @override
-  _DecoyChildState createState() => _DecoyChildState();
+  DecoyChildState createState() => DecoyChildState();
 }
 
-class _DecoyChildState extends State<_DecoyChild> with TickerProviderStateMixin {
+/// The state of the [DecoyChild].
+///
+/// This class is responsible for animating the child widget when the
+/// CupertinoContextMenu is about to open.
+class DecoyChildState extends State<DecoyChild> with TickerProviderStateMixin {
+  /// The point at which the CupertinoContextMenu begins to animate
+  /// into the open position.
+  ///
+  /// A value between 0.0 and 1.0 corresponding to a point in the [builder]'s
+  /// animation. When passing in an animation to [builder] the range before
+  /// [animationOpensAt] will correspond to the animation when the widget is
+  /// pressed and held, and the range after is the animation as the menu is
+  /// fully opening. For an example, see the documentation for [builder].
+  static final double animationOpensAt =
```

Comment:
Is this duplicated now?

Reply:
Check the latest commit, I cleaned up a lot, I made these changes as a result of making `_DecoyChild` public now they are like before.